### PR TITLE
Adding multivalue for enum field.

### PR DESCRIPTION
**Pull request by JaniszM**
Added possibility to check multiple values for one enum field if repeat parameter is set.
Query params addapted to HTTP 1.1 specification.
Some styles changed, please look at this.

### CODE REVIEW

**Comment by sichvoge** [comment]
Hi @JaniszM before doing that, could you `rebase` your commits and sign our [CLA](http://www.mulesoft.org/legal/contributor-agreement.html) [here](https://api-notebook.anypoint.mulesoft.com/notebooks/#380297ed0e474010ff43)

**Comment by dmartinezg** [comment]
Please do not modify the white-spacing rules on Gruntfile.

File: dist/scripts/api-console.js [hunk 1]
```diff
@@ -734,27 +734,43 @@
...
 
-        $scope.isDefault = function (definition) {
+        $scope.isDefault = function(definition) {
           return typeof definition['enum'] === 'undefined' && definition.type !== 'boolean';
         };
 
-        $scope.isEnum = function (definition) {
-          return typeof definition['enum'] !== 'undefined';
+        $scope.isEnum = function(definition) {
+          return (typeof definition['enum'] !== 'undefined' && definition.repeat !== true);
```
Comment (dmartinezg) [inline dist/scripts/api-console.js]:
```
return (typeof definition['enum'] !== 'undefined' && !!definition.repeat);
```

File: dist/scripts/api-console.js [hunk 2]
```diff
@@ -734,27 +734,43 @@
...
         };
 
-        $scope.isBoolean = function (definition) {
+        $scope.isMultiEnum = function(definition) {
+          return (typeof definition['enum'] !== 'undefined' && definition.repeat === true);
```
Comment (dmartinezg) [inline dist/scripts/api-console.js]: return (typeof definition['enum'] !== 'undefined' && !!definition.repeat);

**Comment by dmartinezg** [comment]
There are a lot of syntax changes, which makes hard to understand the functional ones.

Can you help us describing what do you mean by `Query params adapted to HTTP 1.1 specification.
`?

**Comment by JaniszM** [comment]
@sichvoge I singned CLA yesterday :) At least I think so.

@dmartinezg can you provide me formatter for your project? Currently I am using sublime + JsFormat. Then I will reformat everything agian.

For query params please refer to http://tools.ietf.org/html/rfc3986#section-3.4. Good convantion says to not use '+' in query param e.g.: instead doing 'thing=desk+lamp' use 'thing=desk&thing=lamp'.

PS: Why in grunt file do not format spaces?

File: dist/scripts/api-console.js (hunk 1 above)
Comment (JaniszM) [inline dist/scripts/api-console.js]: Done.

File: dist/scripts/api-console.js (hunk 2 above)
Comment (JaniszM) [inline dist/scripts/api-console.js]: Done.

**Comment by dmartinezg** [comment]
@JaniszM, you are right, the project is missing the jshint and jscs presets, we use in other projects.

I will add them in another PR, can you please separate the syntax changes from the other ones?

Also, what effect do the CSS changes have? can you add a screenshot?

**Comment by dmartinezg** [comment]
Can you reset tabs to 2 spaces?

We use .editorconfig (http://editorconfig.org/) to ensure consistent spacing

**Comment by JaniszM** [comment]
Please don't merge this anyway. I found bug related with checkbox when type is boolean. I will check it and commit fix later. @dmartinezg Also I will answer to you later :)

PS: very important test is missing. There isn't any test checking checkboxes :(

**Comment by dmartinezg** [comment]
Thanks for letting us know.

**Comment by JaniszM** [comment]
@dmartinezg  Ok. I fixed bug. Don't bother styles now. I restored all from api-console master and added one style class to multiEnum field. Nothing else is changed. (see attached image)

Regarding spaces. I have set 2 spaces on tab (and space indent), but still can't format code like you do. Please give me some rules of formatting that I can use. These from http://editorconfig.org/ are default and you are using something more I think.

screen shot 2015-12-04 at 15 03 01

**Comment by dmartinezg** [comment]
:+1: Can you please squash commits?

I'll merge it after that.

For spacing, we'll add the right jscs and jshint profiles, and I will fix the differences. We are not formatting using any linter, we format manually according to (missing) jscs rules.

**Comment by JaniszM** [comment]
Done, commits squashed.

**Comment by dmartinezg** [comment]
There still are 8 commits.

**Comment by JaniszM** [comment]
Hmm, you are right. I did some mistake -.- On monday I will squash them all ;)

**Comment by JaniszM** [comment]
Now squashed ;)

**Comment by JaniszM** [comment]
What is status of this PR? Soon I will prepare another one.

**Comment by JaniszM** [comment]
Hey, guys, what about this pull request? It's been a long time since everything is ready to merge ;)

**Comment by dmartinezg** [comment]
Hi @JaniszM, as I said before, this is a beast of a PR, because of all the styling changes, and on top of it, they are not consistent with current style.

Please revert the spacing changes, and leave the PR to a minimum of changes.

@xaka, does this have any impact on the next changes for 1.0?

**Comment by JaniszM** [comment]
Huh, @dmartinezg as I ask before, would be nice to have some formatter rules for this ;) Reverting this stuff can be time consuming and... manually this not have sense :x Please give me some rules, at least formal, and I will try to set up my formatter.

**Comment by JaniszM** [comment]
Ok, done, I think I reverted all spacing changes. Is it fine now?

**Comment by JaniszM** [comment]
Oh My God. I even don't remember what It was for -.- If I find some time I will check it again :o It was so long ago...

**Comment by sichvoge** [comment]
Indeed its been a while :D Sorry for the late response.

**Comment by jarrodek** [comment]
Hi, thank you for the PR. It relates to older version of the console which is no longer maintained.

**Comment by JaniszM** [comment]
@jarrodek At that time it was up to date ;D

**Comment by jarrodek** [comment]
@JaniszM yes, it was :( I am sorry for that.

**Comment by JaniszM** [comment]
No problem, don't worry.

**Pull request closed without merging**